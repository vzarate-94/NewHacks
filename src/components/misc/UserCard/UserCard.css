.user-card{
  display: flex;
  align-items: center;
  padding-left: 10px;
}

.user-card img{
  width: 40px;
  height: 40px;
  border: 1px solid #4d4d63;
  border-radius: 100%;
}

.user-info{
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 0;
  padding: 10px 10px 10px 15px;
}

.user-info * {
  margin: 0;
  color: rgb(215, 215, 215);
}

.user-info h4 {
  margin-bottom: 2px;
}

.user-info p {
  margin-top: 2px;
  font-size: 14px;
  color: rgb(145, 145, 145);
}