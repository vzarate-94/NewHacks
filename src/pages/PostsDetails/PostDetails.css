.post-details {
  height: 100%;
  margin-top: 50px;
  background: #0b0b12;
  padding: 10px;
}

.loading-container {
  padding: 20px;
}

.loading-container div {
  padding: 10px 30px 15px 30px;
}