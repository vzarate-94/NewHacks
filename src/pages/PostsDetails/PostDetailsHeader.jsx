import React from 'react'
import '../../styles/Header.css'


const PostDetailsHeader = (props) => {

    return (
        <div className="hidden-container">
            <div className="header">
                <h3>Post Details</h3>
                <div className="header-buttons">
                    <button onClick={() => props.history.push('/')}>
                        Back
					            </button>
                </div>
            </div>
        </div>
    )
}

export default PostDetailsHeader