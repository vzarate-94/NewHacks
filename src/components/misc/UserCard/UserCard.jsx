import React from 'react'
import './UserCard.css'

const UserCard = (props) => {
    const { user } = props

    return (
        user &&
        <div className="user-card">
            <div className="user-info">
                <h4>{user.name}</h4>
                <h4>Posts</h4>
                
            </div>
        </div>
    )
}

export default UserCard