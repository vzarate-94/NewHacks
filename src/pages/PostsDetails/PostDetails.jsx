import React, { useState, useEffect } from 'react'
import { withRouter, useParams} from 'react-router-dom'
import './PostDetails.css'

import {
  getPostById,
  deletePost
} from '../../services/postService'

import PostCard from '../../components/Post/PostCard'
import Layout from '../../components/Layout/Layout'
import CommentSection from '../../components/Comment/CommentSection'
import PostDetailsHeader from './PostDetailsHeader'
import Animation from '../../components/misc/Animation'

import loading from '../../assets/animation/loading.json'

const PostDetails = (props) => {
  const { id } = useParams()
  const [post, setPost] = useState()
  const [commentArray, setCommentArray] = useState([])

  const handleDeletePost = async (postId) => {
    try {
      await deletePost(postId)
      props.history.push('/')
    } catch (error) {
      throw error
    }
  }

  useEffect(() => {
    const fetchPost = async () => {
        try {
            const post = await getPostById(id)
            setTimeout(() => {
                setPost(post)
                setCommentArray(post.comments)
            }, 1000)
        } catch (error) {
            throw error
        }
    }
    fetchPost()
    return () => { setPost(null) }
}, [id])

  return (
    <Layout {...props}>
      <div className='layout'>
      <PostDetailsHeader {...props} />
        <div className='post-details'>
          {post? 
          <>
            <PostCard
              post={post}
              currentUser={props.currentUser}
              handleDeletePost={handleDeletePost}
            />
            <CommentSection
              post={post}
              setPost={setPost}
              currentUser={props.currentUser}
              commentArray={commentArray}
              setCommentArray={setCommentArray}
            />
          </>
          :
          <div className='loading-container'>
            <div>
              <Animation animData={loading}></Animation>
            </div>
          </div>
          }

        </div>
      </div>
    </Layout>
  )
}

export default withRouter(PostDetails)