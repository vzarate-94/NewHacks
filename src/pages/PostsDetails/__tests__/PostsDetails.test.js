import React from 'react';
import ReactDOM from 'react-dom';
import PostDetails from '../PostDetails';
import UserCard from '../../../components/misc/UserCard/UserCard';
import PostDetailsHeader from '../PostDetailsHeader';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('UserCard Component render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<UserCard></UserCard>, div);
});

it('PostDetailsHeader renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PostDetailsHeader />, div);
});

it('matches Snapshot', () => { 
  const tree = renderer.create(<PostDetailsHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});