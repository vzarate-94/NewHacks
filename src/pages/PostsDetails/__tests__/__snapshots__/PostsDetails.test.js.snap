// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<div
  className="hidden-container"
>
  <div
    className="header"
  >
    <h3>
      Post Details
    </h3>
    <div
      className="header-buttons"
    >
      <button
        onClick={[Function]}
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
